solvation_analysis/analysis.py

```diff
@@ -0,0 +1,127 @@
+# from abc import ABC
+from abc import ABC
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.analysis.rdf import InterRDF
+from MDAnalysis.analysis import distances
+import numpy as np
+from solvation_analysis.rdf_parser import identify_solvation_cutoff
+from solvation_analysis.analysis_library import (
+    _CoordinationNumbers,
+    _IonPairing,
+    _IonSpeciation,
+)
+
+
+def some_function():
+    return
+
+
+class Solute(AnalysisBase):
+    """
+    The core class of the solvation module.
+    """
+
+    def __init__(
+        self, solute, solvents, radii=None, rdf_kernel=None, kernel_kwargs=None, **kwargs
+    ):
+        """
+        Parameters
+        ----------
+        solute (AtomGroup): an atom group  # TODO: force to atom group
+        solvents (dict): a dictionary of names and atom groups.
+            e.g. {"name_1": solvent_group_1, "name_2": solvent_group_2, ...}
+        radii (dict): an optional dictionary of solvation radii, any radii not
+            given will be calculated. e.g. {"name_2": radius_2, "name_5": radius_5}
+        rdf_kernel (func): this function must take rdf bins and data as input and return
+            a solvation radius as output. e.g. rdf_kernel(bins, data) -> 3.2. By default,
+            the rdf_kernel is solvation_analysis.rdf_parser.identify_solvation_cutoff.
+        kernel_kwargs: kwargs passed to rdf_kernel
+        kwargs: kwargs passed to AnalysisBase
+        """
+        super(Solute, self).__init__(solute.universe.trajectory, **kwargs)
+        self.radii = {} if radii is None else radii
+        self.kernel = identify_solvation_cutoff if rdf_kernel is None else rdf_kernel
+        # if not kernel:
+        #     self.kernel = identify_solvation_cutoff
+        self.kernel_kwargs = {} if kernel_kwargs is None else radii
+        # if not kernel_kwargs:
+        #     self.kernel_kwargs = {}
+
+        self.solute = solute
+        self.solvents = solvents
+        self.u = self.solute.universe
+        self.rdf_plots = {}
+        self.rdf_data = {}
+        self.ion_speciation = None
+        self.ion_pairing = None
+        self.coordination_numbers = None
+        self.solvation_frames = []
+
+    @classmethod
+    def _plot_solvation_radius(cls, bins, data, radius):
+        fig, ax = plt.subplots()
+        ax.plot(bins, data, "b-", label="rdf")
+        ax.axvline(radius, color="r", label="solvation radius")
+        ax.set_xlabel("Radial Distance (A)")
+        ax.set_ylabel("Probability Density")
+        ax.legend()
+        return fig, ax
+
+    def run_prepare(self):
```

Comment:
You'll need to add docstring here to make sure that folks understand what this does (since it's user facing), also so that devs know the difference between `run_prepare` and `_prepare`. In my opinion it might be worth going for something a bit more descriptive (or at least different from `_prepare`) so there's no confusion as to what this does.

Reply:
This is merged in the most recent commit!